### PR TITLE
[upmeter] Fix garbage collection across probes

### PR DESCRIPTION
**Pull request by shvgn**
## Description

- Fix garbage collection of probe objects
- Use static names for probe objects to reduce metrics cardinality

## Why do we need it, and what problem does it solve?

- Reduces the amount of garbage objects produced by upmeter agents 
- Improves cleanup of these object
- Reduces the cardinality in kube state metrics by using static object names

## What is the expected result?

Certificates and pods produced by probes should not be left in clusters

## Checklist

- [x] The code is covered by unit tests.
- [ ] e2e tests are passed.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: upmeter
type: fix
summary: Fixed garbage collection for control-plane probes
```

## Cleaning objects from previous version

```sh
kubectl -n d8-upmeter get cm -o=name | grep upmeter-basic | xargs -- kubectl -n d8-upmeter delete &>/dev/null

kubectl -n d8-upmeter get certificates -o=name | grep upmeter-cm | xargs -- kubectl -n d8-upmeter delete &>/dev/null

kubectl -n d8-upmeter get secret -o=name | grep upmeter-cm | xargs -- kubectl -n d8-upmeter delete &>/dev/null

kubectl -n d8-upmeter get po -o=name | grep upmeter-probe-basic | xargs -- kubectl -n d8-upmeter delete &>/dev/null

kubectl -n d8-upmeter get deploy -o=name | grep upmeter-controller-manager | xargs -- kubectl -n d8-upmeter delete &>/dev/null
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

File: modules/500-upmeter/hooks/clean_probe_garbage.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
+// Copyright 2021 Flant JSC
```
Comment (z9r5) [inline modules/500-upmeter/hooks/clean_probe_garbage.go:1]:
```suggestion
// Copyright 2022 Flant JSC
```

**Review by z9r5 — APPROVED** [review]
LGTM, but usually we use only english in internal (dev.) docs. Tell me when everything is ready for translation.

**Comment by shvgn** [comment]
Oh. I see. The doc was in Russian initially, so I just added stuff and formatted it accidentally. 

As for me, it is not well-written. I would do another PR for the doc right after we merge this one. With that, I will verify the text. I see mistakes in it.

**Comment by shvgn** [comment]
@z9r5 removed the doc changes to another PR, please take a look #2017

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

**Comment by z9r5** [comment]
Cherry-picked into release-1.34

**Pull request merged**